integration/spark2/src/main/scala/org/apache/spark/sql/execution/command/schema/CarbonAlterTableRenameCommand.scala

```diff
@@ -77,6 +78,11 @@ private[sql] case class CarbonAlterTableRenameCommand(
     var oldCarbonTable: CarbonTable = null
     oldCarbonTable = metastore.lookupRelation(Some(oldDatabaseName), oldTableName)(sparkSession)
       .asInstanceOf[CarbonRelation].carbonTable
+
+    if (CarbonUtil.validateFeatureForDatamap(oldCarbonTable, FeaturesList.ALTER_RENAME)) {
```

Comment:
After you handle previous comments, I think this line will become:
` if (oldCarbonTable.canAccept(TableOperation.ALTER_RENAME))`